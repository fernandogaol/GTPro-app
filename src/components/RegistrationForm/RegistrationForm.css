.RegistrationForm label {
  padding: 2px;
  font-size: 1.1rem;
}
.register-button {
  background: rgb(50, 92, 230);
  color: #fff;
}
.full_name,
.user_name,
.password {
  text-align: left;
}
#RegistrationForm__password,
#RegistrationForm__user_name,
#RegistrationForm__full_name {
  margin: 10px 0;
}

.red {
  color: red;
  padding: 10px;
  background: #fff;
  margin: 15px 0;
}
@media screen and (min-width: 480px) {
}
