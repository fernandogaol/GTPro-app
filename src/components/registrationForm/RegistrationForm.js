import React, { Component } from 'react';

export default class RegistrationForm extends Component {
  render() {
    return (
      <form className='RegistrationForm'>
        {/* <div role='alert'>{error && <p className='red'>{error}</p>}</div> */}
        <div className='full_name'>
          <label htmlFor='RegistrationForm__full_name'>
            Full name <required />
          </label>
          <input
            name='full_name'
            type='text'
            requinput
            id='RegistrationForm__full_name'
          ></input>
        </div>
        <div className='user_name'>
          <label htmlFor='RegistrationForm__user_name'>
            User name <required />
          </label>
          <input
            name='user_name'
            type='text'
            required
            id='RegistrationForm__user_name'
          ></input>
        </div>
        <div className='password'>
          <label htmlFor='RegistrationForm__password'>
            Password <required />
          </label>
          <input
            name='password'
            type='password'
            required
            id='RegistrationForm__password'
          ></input>
        </div>
        <div className='nick_name'>
          <label htmlFor='RegistrationForm__nick_name'>Nickname</label>
          <input
            name='nick_name'
            type='text'
            id='RegistrationForm__nick_name'
          ></input>
        </div>
        <button type='submit'>Register</button>
      </form>
    );
  }
}
