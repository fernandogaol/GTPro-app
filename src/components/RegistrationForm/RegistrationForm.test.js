import React from 'react';
import renderer from 'react-test-renderer';
import RegistrationForm from './RegistrationForm';

it('renders correctly', () => {
  const tree = renderer.create(<RegistrationForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
