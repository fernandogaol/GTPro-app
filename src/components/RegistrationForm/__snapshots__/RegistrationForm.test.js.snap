// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="full_name"
  >
    <label
      htmlFor="RegistrationForm__full_name"
    >
      Full name
    </label>
    <input
      className="Input "
      id="RegistrationForm__full_name"
      name="full_name"
      placeholder="Joe Doe"
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm__user_name"
    >
      Username
    </label>
    <input
      className="Input "
      id="RegistrationForm__user_name"
      name="user_name"
      placeholder="JoeDoe"
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="RegistrationForm__password"
    >
      Password
    </label>
    <input
      className="Input "
      id="RegistrationForm__password"
      name="password"
      placeholder="Password123"
      type="password"
    />
  </div>
  <button
    className="Button register-button"
    type="submit"
  >
    Register
  </button>
</form>
`;
